lib/sqlalchemy/dialects/oracle/cx_oracle.py

```diff
@@ -208,32 +234,34 @@ def creator():
 pooling and Oracle Database additionally uses DRCP::
 
     with engine.connect() as conn:
-        print(conn.scalar("select 1 FROM dual"))
+        print(conn.scalar("select 1 from dual"))
 
 .. _cx_oracle_unicode:
 
 Unicode
 -------
 
 As is the case for all DBAPIs under Python 3, all strings are inherently
-Unicode strings.   In all cases however, the driver requires an explicit
+Unicode strings. In all cases however, the driver requires an explicit
 encoding configuration.
 
 Ensuring the Correct Client Encoding
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 The long accepted standard for establishing client encoding for nearly all
-Oracle related software is via the `NLS_LANG <https://www.oracle.com/database/technologies/faq-nls-lang.html>`_
-environment variable.   cx_Oracle like most other Oracle drivers will use
-this environment variable as the source of its encoding configuration.  The
-format of this variable is idiosyncratic; a typical value would be
-``AMERICAN_AMERICA.AL32UTF8``.
+Oracle Database related software is via the `NLS_LANG
+<https://www.oracle.com/database/technologies/faq-nls-lang.html>`_ environment
+variable.  Older versions of cx_Oracle use this environment variable as the
+source of its encoding configuration.  The format of this variable is
+Territory_Country.CharacterSet; a typical value would be
+``AMERICAN_AMERICA.AL32UTF8``.  cx_Oracle version 8 and later use the character
+set "UTF-8" by default, and ignore the character set component of NLS_LANG.
 
-The cx_Oracle driver also supports a programmatic alternative which is to
-pass the ``encoding`` and ``nencoding`` parameters directly to its
-``.connect()`` function.  These can be present in the URL as follows::
+The cx_Oracle driver also supported a programmatic alternative which is to pass
```

Comment:
q: are they still supported but deprecated or were completely removed in cx-oracle?

if so maybe we could remove the reference to encoding/neconding from here and other sections

Reply:
We eased the change in gradually - which now gets hard to explain.  It wasn't until python-oracledb 2.0 that using them throws a warning.  I think we can leave the cx_Oracle doc as-is.